demos/test-app/lib.rs

```diff
@@ -106,160 +74,108 @@ pub enum AlternativeOriginsMode {
-            BASE64.encode(&certificate),
-            BASE64.encode(&serializer.into_inner())
-        ),
+fn static_headers() -> Vec<(String, String)> {
+    vec![("Access-Control-Allow-Origin".to_string(), "*".to_string())]
+}
+
+// Assets
+static ASSET_DIR: Dir<'_> = include_dir!("$CARGO_MANIFEST_DIR/dist");
+
+fn fixup_html(html: &str) -> String {
+    let canister_id = api::id();
+
+    // the string we are replacing here is inserted by vite during the front-end build
+    html.replace(
+        r#"<script type="module" crossorigin src="/index.js"></script>"#,
+        &format!(r#"<script data-canister-id="{canister_id}" type="module" crossorigin src="/index.js"></script>"#).to_string(),
     )
 }
 
 #[init]
 #[post_upgrade]
-pub fn init_assets() {
-    ASSETS.with(|a| {
-        ASSET_HASHES.with(|ah| {
-            let mut assets = a.borrow_mut();
-            let mut asset_hashes = ah.borrow_mut();
-            for (path, content, content_type) in get_assets() {
-                asset_hashes.insert(path, sha2::Sha256::digest(content).into());
-                let mut headers = vec![];
-                headers.push((
-                    "Content-Type".to_string(),
-                    content_type.to_mime_type_string(),
-                ));
-                assets.insert(path, (headers, content.to_vec()));
-            }
-            update_root_hash(&asset_hashes);
-        });
-    });
-}
-
-fn update_root_hash(a: &RbTree<&'static str, [u8; 32]>) {
-    let prefixed_root_hash = labeled_hash(b"http_assets", &a.root_hash());
-    api::set_certified_data(&prefixed_root_hash[..]);
+pub fn init() {
+    init_assets(EMPTY_ALTERNATIVE_ORIGINS.to_string());
 }
 
-lazy_static! {
-    // The full content of the index.html, after the canister ID (and script tag) have been
-    // injected
-    static ref INDEX_HTML_STR: String = {
-        let canister_id = api::id();
-        let index_html = include_str!("dist/index.html");
+/// Collect all the assets from the dist folder.
+fn init_assets(alternative_origins: String) {
+    let mut assets = collect_assets(&ASSET_DIR, IncludeSubdirs, Some(fixup_html));
+    assets.push(Asset {
+        url_path: ALTERNATIVE_ORIGINS_PATH.to_string(),
+        content: alternative_origins.as_bytes().to_vec(),
+        encoding: ContentEncoding::Identity,
+        content_type: ContentType::JSON,
+    });
 
-        // the string we are replacing here is inserted by vite during the front-end build
-        let index_html = index_html.replace(
-            r#"<script type="module" crossorigin src="/index.js"></script>"#,
-            &format!(r#"<script data-canister-id="{canister_id}" type="module" crossorigin src="/index.js"></script>"#).to_string()
-        );
-        index_html
-    };
+    // convenience asset to have an url to point to when testing with the redirect alternative origins behaviour
+    assets.push(Asset {
+        url_path: "/.well-known/evil-alternative-origins".to_string(),
```

Comment:
nit: maybe use a constant, say `EVIL_ALTERNATIVE_ORIGINS_PATH` -- currently it is not obvious that this value is different from  `ALTERNATIVE_ORIGINS_PATH` used in line 158.